File: CBHTTP/Models/Credentials.swift
// Copyright (c) 2017-2019 Coinbase Inc. See LICENSE

import Foundation

/// HTTP basic authentication credentials
public struct Credentials {
    /// HTTP basic authentication username
    public let username: String

    /// HTTP basic authentication password
    public let password: String

    internal var basicAuth: String? {
        let credentialString = "\(username):\(password)"

        guard let data = credentialString.data(using: .utf8) else {
            return nil
        }

        return "Basic \(data.base64EncodedString())"
    }
}

File: android/http/src/main/java/com/coinbase/wallet/http/models/Credentials.kt
package com.coinbase.wallet.http.models

import com.coinbase.wallet.core.extensions.base64EncodedString

/**
 * HTTP basic authentication credentials
 *
 * @property username HTTP basic authentication username
 * @property password HTTP basic authentication password
 */
data class Credentials(val username: String, val password: String) {
    /**
     * Properly encoded and formatted HTTP basic authentication header value
     */
    val basicAuth: String?
        get() {
            val credentialString = "$username:$password"
            val data = credentialString.toByteArray(Charsets.UTF_8)

            return "Basic ${data.base64EncodedString()}"
        }

    companion object
}
